src/gator/core/models/timetable.py

```diff
@@ -13,131 +13,116 @@
+                return f'{self.year}5'
 
     def __str__(self) -> str:
-        """Return a string representation of this session.
-
-        The output string is the session code. See `code` for more
-        information.
-        """
+        """Return a string representation of the session."""
         return self.code
 
     @classmethod
-    def parse(cls, session_code: Union[str, int]) -> 'Session':
-        """Return an instance Session representing the given session code.
-
-        Raise a ValueError if the session code is not formatted properly.
+    def from_code(cls, code: str) -> 'Session':
+        """Return a session from its canonical representation.
 
         Args:
-            session_code: The session code to parse as a string or integer.
-                If the session code is an integer, it will be converted to a
-                string and zero-padded to the left if necessary. E.g. if
-                509 is passed, it will be converted to '00509' which corresponds
-                to the fall session of the year 50.
+            code: The canonical representation of the session.
 
-        Examples:
-            >>> Session.parse(509) == Session(year=50, season='fall')
-            True
-            >>> Session.parse(20201) == Session(year=2020, season='winter')
-            True
-            >>> Session.parse('20205') == Session(year=2020, season='summer')
-            True
+        Raises:
+            ValueError: If the given code is not a valid session code.
         """
-        # Ensure that the session code is a string and zero-padded to the left
-        session_code = str(session_code).zfill(5)
-
-        if len(session_code) != 5:
-            raise ValueError(
-                f'invalid session code ("{session_code}"): expected string of length 5'
-                f', not {len(session_code)}')
-        elif not session_code.isnumeric():
-            raise ValueError(
-                f'invalid session code ("{session_code}"): expected numeric string')
-        elif int(session_code[-1]) not in cls._INVERSE_SEASON_MAP:
-            raise ValueError(
-                f'invalid session code ("{session_code}"): expected code to end in '
-                f'one of {list(cls._INVERSE_SEASON_MAP.keys())}, not {session_code[-1]}')
+        if re.match(r'^\d{4}[9|1]$', code):
+            return cls(int(code[:4]), 'regular', 'first' if code[4] == '9' else 'second')
+        elif re.match(r'^\d{4}9-\d{4}1$', code):
+            return cls(int(code[:4]), 'regular', 'whole')
+        elif re.match(r'^\d{4}5(F|S)$', code):
+            return cls(int(code[:4]), 'summer', 'first' if code[4] == 'F' else 'second')
+        elif re.match(r'^\d{4}5$', code):
+            return cls(int(code[:4]), 'summer', 'whole')
         else:
-            return Session(
-                int(session_code[:4]),
-                cls._INVERSE_SEASON_MAP[int(session_code[-1])])
+            raise ValueError(f'Invalid session code: {code}')
 
```

Comment:
Any idea what happened to the `human_str` method? We rely on it in a few different places so I'd prefer to keep it.